### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by Masouma96**
Use validations for models.
Write unit tests for models.

Make test for each files

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# end with _spec.rb. You can configure this pattern with the --pattern
+# option on the command line or in ~/.rspec, .rspec or `.rspec-local`.
+#
+# The following line is provided for convenience purposes. It has the downside
+# of increasing the boot-up time by auto-requiring all files in the support
+# directory. Alternatively, in the individual `*_spec.rb` files, manually
+# require only the support files necessary.
+#
+# Dir[Rails.root.join('spec', 'support', '**', '*.rb')].sort.each { |f| require f }
+
```
Comment (t-yanick) [inline spec/rails_helper.rb:24]: - [ ] Great job working on the project. But your `spec` folder is missing the test. I cannot find any test in the `spec` folder. Please kindly check and update this issue. You should have unit tests for `User`, `Post`, `Comment`, and `Like` model. Check example below:

**_Model specs has **Validation** tests Example:_**
```
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```

- [ ] Add unit specs for all of your models' methods and validations.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @Masouma864 , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All linters checks passed
:heavy_check_mark: Good commit messages, but consider capitalizing first letters

There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] I am requesting changes here because your ReadMe file does not show in Files Changed.
- Kindly add the steps to run your project locally. I cannot find command like 
_rails db:create
rails db:migrate_
Before starting the server

- [ ] Kindly make your PR descrition more descriptive rather than the 2 sentences you currently have. Please also make the PR title more descriptive

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Capitalizing the first letters of your commit messages

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/model/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    user = User.new(name: 'kareem', photo: 'www.photo', bio: 'this is bio', posts_counter: 0)
+    post = Post.create(title: 'post 1', text: 'here is text', comments_counter: 0, likes_counter: 0, user:)
+    Comment.create(text: 'first comment', user:, post:)
+    Comment.create(text: 'second comment', user:, post:)
+  end
+
+  it 'should update comments counter' do
+    expect(subject.update_comments_counter).to eq true
+    expect(subject.post.comments_counter).to eq 2
+  end
```
Comment (Munsa1) [inline spec/model/comment_spec.rb]: - [ ] As requested by te previous reviewer, kindly correct your tests as they are all failing
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Masouma864, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @Masouma864 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image

## Highlights 
:+1: no linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**